src/controls/QskTextAreaSkinlet.cpp

```diff
@@ -0,0 +1,103 @@
+/******************************************************************************
+ * QSkinny - Copyright (C) The authors
+ *           SPDX-License-Identifier: BSD-3-Clause
+ *****************************************************************************/
+
+#include "QskTextAreaSkinlet.h"
+#include "QskTextArea.h"
+
+using Q = QskTextArea;
+
+QskTextAreaSkinlet::QskTextAreaSkinlet( QskSkin* skin )
+    : Inherited( skin )
+{
+    setNodeRoles( { PanelRole, TextPanelRole, PlaceholderTextRole } );
+}
+
+QskTextAreaSkinlet::~QskTextAreaSkinlet()
+{
+}
+
+QRectF QskTextAreaSkinlet::subControlRect( const QskSkinnable* skinnable,
+    const QRectF& contentsRect, QskAspect::Subcontrol subControl ) const
+{
+    if ( subControl == Q::Panel )
+        return contentsRect;
+
+    if ( subControl == Q::TextPanel )
+        return skinnable->subControlContentsRect( contentsRect, Q::Panel );
+
+    if ( subControl == Q::PlaceholderText )
+    {
+        const auto textArea = static_cast< const QskTextArea* >( skinnable );
+        if( textArea->text().isEmpty() )
+            return subControlRect( skinnable, contentsRect, Q::Text );
+
+        return QRectF();
+    }
+
+    return Inherited::subControlRect( skinnable, contentsRect, subControl );
+}
+
+QSGNode* QskTextAreaSkinlet::updateSubNode(
+    const QskSkinnable* skinnable, quint8 nodeRole, QSGNode* node ) const
+{
+    const auto textArea = static_cast< const QskTextArea* >( skinnable );
+
+    switch ( nodeRole )
+    {
+        case PanelRole:
+        {
+            return updateBoxNode( skinnable, node, Q::Panel );
+        }
+
+        case PlaceholderTextRole:
+        {
+            const auto text = effectivePlaceholderText( textArea );
+            if ( text.isEmpty() )
+                return nullptr;
+
+            const auto subControl = Q::PlaceholderText;
+
+            QskSkinHintStatus status;
+
+            auto options = skinnable->textOptionsHint( subControl, &status );
+            if ( !status.isValid() )
+                options.setElideMode( Qt::ElideRight );
```

Comment:
Does the elide mode make much sense here ? A text area is multi line and I would expect, that the placeholder text would be fitted into the viewport. ( -> QskTextOptions::Wrap )

Reply:
But wait - you are probably right. One also wouldn't expect to have to scroll the placeholder text.